### PR TITLE
Databricks CICD content verifying and fixing

### PR DESCRIPTION
**Pull request by cchenshu**
# Type of PR

- Code changes
- Test changes
- CI-CD changes

## Purpose

- The test failed when using [./devops/notebook-pipelines.yml](https://github.com/Azure-Samples/modern-data-warehouse-dataops/blob/anuj/sample-archiving/single_tech_samples/databricks/databricks_ci_cd/devops/notebook-pipelines.yml) as build pipeline. Failed at "Execute Nutter".
**Reason: jobs run more than 120s (default timeout)**
**Fix: add timeout parameter and set it to 600s**

- For Spark Python Approach, the test failed when using [./devops/spark-python-pipelines.yml](https://github.com/Azure-Samples/modern-data-warehouse-dataops/blob/anuj/sample-archiving/single_tech_samples/databricks/databricks_ci_cd/devops/spark-python-pipelines.yml) as build pipeline. Failed at  "Launch integration on test", and the reason is " HTTPError: 400 Client Error: Bad Request for url: Response from server: { 'error_code': 'INVALID_PARAMETER_VALUE','message': 'Invalid python file reference: ''spark_python_jobs/tests/integration/main_test.py.Please visit ''the Databricks user guide for supported python references'}"
**Reason: Databricks versions has updated**
**Fix:  
          - update deployment json file
          - update project.json file 
          - update spark and dbx version
          - update dbx deploy command
          - update dbx launch command**
         
- Fix typos for DATABRICKS_JON_NAME

## Author pre-publish checklist

- [ ] Added test to prove my fix is effective or new feature works
- [ ] No PII in logs
- [ ] Made corresponding changes to the documentation

## Validation steps

- ...

## Issues Closed or Referenced

- Closes #issue_number

- References #issue_number

### CODE REVIEW

File: single_tech_samples/databricks/databricks_ci_cd/devops/template/test-spark-python-job.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ jobs:
 
         - script: |
             echo "dbx launch job $DATABRICKS_JON_NAME ..."
-            dbx launch --job=$DATABRICKS_JON_NAME --as-run-submit --trace
+            dbx launch $DATABRICKS_JON_NAME --as-run-submit --trace
```
Comment (thurstonchen) [inline single_tech_samples/databricks/databricks_ci_cd/devops/template/test-spark-python-job.yml:69]: Hi @chenshu,
The PR looks good! Thanks for your help to make this single tech sample up-to-date!
Just a minor issue, there's an legacy typo in variable name _$DATABRICKS_**JON**_NAME_, would you please help to correct them in all relevant occurrences, thanks a lot! :- )
Best,
Thurston

Comment (cchenshu) [reply single_tech_samples/databricks/databricks_ci_cd/devops/template/test-spark-python-job.yml]: Done! Thank you!

**Review by thurstonchen — APPROVED** [review]

**Comment by devlace** [comment]
Consider archiving existing Databricks sample in favor of upgrading.

**Comment by devlace** [comment]
Looks good to me!

**Review by devlace — APPROVED** [review]

**Pull request merged**